### PR TITLE
Normative: Add the `at` method to Array, TypedArray, and String

### PR DESCRIPTION
**Pull request by syg**

### CODE REVIEW

File: spec.html [hunk 1]
```diff
@@ -31728,6 +31744,21 @@ <h1>Properties of the Array Prototype Object</h1>
         <p>The Array prototype object is specified to be an Array exotic object to ensure compatibility with ECMAScript code that was created prior to the ECMAScript 2015 specification.</p>
       </emu-note>
```
Comment (loganfsmyth) [inline spec.html:31745]: Is this busted? There's no `</emu-clause>` on this and the indentation is off.

File: spec.html [hunk 2]
```diff
@@ -31728,6 +31744,21 @@ <h1>Properties of the Array Prototype Object</h1>
...
 
+      <emu-clause id="sec-array.prototype.at">
+        <h1>Array.prototype.at ( _index_ )</h1>
+        <emu-alg>
+          1. Let _O_ be ? ToObject(*this* value).
+          1. Let _len_ be ? LengthOfArrayLike(_O_).
+          1. Let _relativeIndex_ be ? ToInteger(_index_).
```
Comment (loganfsmyth) [inline spec.html:31752]: Is this `ToIntegerOrInfinity` these days?

File: spec.html [hunk 3]
```diff
@@ -28503,6 +28503,22 @@ <h1>Properties of the String Prototype Object</h1>
         1. Throw a *TypeError* exception.
       </emu-alg>
 
+      <emu-clause id="sec-string.prototype.at">
+        <h1>String.prototype.at ( _index_ )</h1>
+        <emu-alg>
+          1. Let _O_ be ? RequireObjectCoercible(*this* value).
+          1. Let _S_ be ? ToString(_O_).
+          1. Let _len_ be the length of _S_.
+          1. Let _relativeIndex_ be ? ToInteger(_index_).
```
Comment (ljharb) [inline spec.html:28512]:
```suggestion
          1. Let _relativeIndex_ be ? ToIntegerOrInfinity(_index_).
```

File: spec.html (hunk 1 above)
Comment (ljharb) [reply spec.html:31745]: GitHub can’t really handle expanding diffs on the web on a file as large as the spec.

File: spec.html (hunk 2 above)
Comment (ljharb) [reply spec.html:31752]:
```suggestion
          1. Let _relativeIndex_ be ? ToIntegerOrInfinity(_index_).
```

File: spec.html [hunk 4]
```diff
@@ -33331,6 +33363,22 @@ <h1>get %TypedArray%.prototype.byteOffset</h1>
         </emu-alg>
       </emu-clause>
 
+      <emu-clause id="sec-%typedarray%.prototype.at">
+        <h1>%TypedArray%.prototype.at ( _index_ )</h1>
+        <emu-alg>
+          1. Let _O_ be the *this* value.
+          1. Perform ? ValidateTypedArray(_O_).
+          1. Let _len_ be _O_.[[ArrayLength]].
+          1. Let _relativeIndex_ be ? ToInteger(_index_).
```
Comment (ljharb) [inline spec.html:33372]:
```suggestion
          1. Let _relativeIndex_ be ? ToIntegerOrInfinity(_index_).
```

**Review by ljharb — APPROVED** [review]

**Review by lastthyme — APPROVED** [review]

**Comment by bakkot** [comment]
We did this in #2493 instead, I guess.

**Comment by ljharb** [comment]
whoops

**Pull request closed without merging**